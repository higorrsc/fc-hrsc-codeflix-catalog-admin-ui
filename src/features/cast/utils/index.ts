import { CastMemberType } from '../types';

export function getCastMemberTypeOptions() {
  return Object.values(CastMemberType)
    .filter((value) => typeof value === 'number')
    .map((value) => ({
      value,
      label:
        CastMemberType[value as CastMemberType].charAt(0) +
        CastMemberType[value as CastMemberType].slice(1).toLowerCase(),
    }));
}
