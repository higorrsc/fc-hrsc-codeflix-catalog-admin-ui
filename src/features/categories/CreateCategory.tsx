import { Box, Paper, Typography } from '@mui/material';
import { useState } from 'react';
import { Category, emptyCategory } from './categorySlice';
import { CategoryForm } from './components/CategoryForm';

export const CreateCategory = () => {
  const [category, setCategory] = useState<Category>(emptyCategory);
  const [isDisabled, setIsDisabled] = useState(false);

  const handleChange = (e: any) => {};
  const handleToggle = (e: any) => {};

  return (
    <Box>
      <Paper>
        <Box p={2}>
          <Box mb={2}>
            <Typography variant='h4'>Create Category</Typography>
          </Box>
        </Box>
        <CategoryForm
          category={category}
          isDisabled={isDisabled}
          isLoading={false}
          handleSubmit={() => {}}
          handleChange={handleChange}
          handleToggle={handleToggle}
        />
      </Paper>
    </Box>
  );
};
