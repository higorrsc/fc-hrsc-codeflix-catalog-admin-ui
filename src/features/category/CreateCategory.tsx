import { useSnackbar } from 'notistack';
import { useEffect, useState } from 'react';
import { Page } from '../../components/Page';
import { initialState, useCreateCategoryMutation } from './categorySlice';
import { CategoryForm } from './components/CategoryForm';
import { Category } from './types';

export const CreateCategory = () => {
  const [createCategory, status] = useCreateCategoryMutation();
  const [isDisabled, setIsDisabled] = useState(false);
  const [categoryState, setCategoryState] = useState<Category>(initialState);

  const { enqueueSnackbar } = useSnackbar();

  const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
    const { name, value } = e.target;
    setCategoryState({ ...categoryState, [name]: value });
  };

  const handleToggle = (e: React.ChangeEvent<HTMLInputElement>) => {
    const { name, checked } = e.target;
    setCategoryState({ ...categoryState, [name]: checked });
  };

  async function handleSubmit(e: React.FormEvent<HTMLFormElement>) {
    e.preventDefault();
    await createCategory(categoryState);
  }

  useEffect(() => {
    if (status.isSuccess) {
      enqueueSnackbar('Category created successfully!', { variant: 'success' });
      setIsDisabled(true);
    }
    if (status.error) {
      enqueueSnackbar('Error creating category!', { variant: 'error' });
    }
  }, [status, enqueueSnackbar]);

  return (
    <Page title='Create Category'>
      <CategoryForm
        category={categoryState}
        isDisabled={isDisabled}
        isLoading={false}
        handleSubmit={handleSubmit}
        handleChange={handleChange}
        handleToggle={handleToggle}
      />
    </Page>
  );
};
