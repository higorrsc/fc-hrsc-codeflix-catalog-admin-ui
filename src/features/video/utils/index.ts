import { Video, VideoPayload } from 'src/features/video/types';

export const mapVideoToForm = (video: Video): VideoPayload => {
  return {
    id: video.id,
    title: video.title,
    description: video.description,
    year_launched: video.year_launched,
    opened: video.opened,
    rating: video.rating,
    duration: video.duration,
    genres_id: video.genres?.map((genre) => genre.id),
    categories_id: video.categories?.map((category) => category.id),
    cast_members_id: video.cast_members?.map((cast_member) => cast_member.id),
  };
};
