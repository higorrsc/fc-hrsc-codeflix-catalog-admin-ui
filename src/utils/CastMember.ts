import { CastMemberType } from '../types/CastMember';

const castMemberTypeLabelMap: Record<CastMemberType, string> = {
  [CastMemberType.ACTOR]: 'Actor',
  [CastMemberType.DIRECTOR]: 'Director',
};

export function getCastMemberTypeLabel(type: CastMemberType): string {
  return castMemberTypeLabelMap[type] || 'Unknown';
}

export const castMemberTypeOptions = [
  { value: CastMemberType.ACTOR, label: 'Actor' },
  { value: CastMemberType.DIRECTOR, label: 'Director' },
];
