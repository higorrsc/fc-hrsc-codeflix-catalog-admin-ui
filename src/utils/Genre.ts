import { Category } from '../types/Category';
import { Genre } from '../types/Genre';

export const mapGenreToForm = (genre: Genre) => {
  return {
    id: genre.id,
    name: genre.name,
    is_active: genre?.is_active,
    categories_id: genre.categories?.map((category: Category) => category.id),
  };
};
