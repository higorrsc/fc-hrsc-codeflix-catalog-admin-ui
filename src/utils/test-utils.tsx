import type { RenderOptions } from '@testing-library/react';
import { render } from '@testing-library/react';
import React, { PropsWithChildren } from 'react';
import { Provider } from 'react-redux';

import { ThemeProvider } from '@mui/system';
import { configureStore } from '@reduxjs/toolkit';
import { SnackbarProvider } from 'notistack';
import { BrowserRouter } from 'react-router-dom';
import type { RootState } from '../app/store';
import { appTheme } from '../config/theme';
import { apiSlice } from '../features/api/apiSlice';
import { castMembersApiSlice } from '../features/cast/castMemberSlice';
import { categoriesApiSlice } from '../features/categories/categorySlice';
// As a basic setup, import your same slice reducers

// This type interface extends the default options for render from RTL, as well
// as allows the user to specify other things such as initialState, store.
interface ExtendedRenderOptions extends Omit<RenderOptions, 'queries'> {
  preloadedState?: Partial<RootState>;
  store?: ReturnType<typeof configureStore>;
}

export function renderWithProviders(
  ui: React.ReactElement,
  extendedRenderOptions: ExtendedRenderOptions = {}
) {
  const {
    // Automatically create a store instance if no store was passed in
    store = configureStore({
      reducer: {
        [categoriesApiSlice.reducerPath]: apiSlice.reducer,
        // @ts-ignore
        [castMembersApiSlice.reducerPath]: apiSlice.reducer,
      },
      // Adding the api middleware enables caching, invalidation, polling,
      // and other useful features of `rtk-query`.
      middleware: (getDefaultMiddleware) =>
        getDefaultMiddleware().concat(apiSlice.middleware),
    }),
    ...renderOptions
  } = extendedRenderOptions;

  const Wrapper = ({ children }: PropsWithChildren) => (
    <React.StrictMode>
      <BrowserRouter>
        <Provider store={store}>
          <ThemeProvider theme={appTheme}>
            <SnackbarProvider maxSnack={3}>{children}</SnackbarProvider>
          </ThemeProvider>
        </Provider>
      </BrowserRouter>
    </React.StrictMode>
  );

  // Return an object with the store and all of RTL's query functions
  return {
    store,
    ...render(ui, { wrapper: Wrapper, ...renderOptions }),
  };
}
