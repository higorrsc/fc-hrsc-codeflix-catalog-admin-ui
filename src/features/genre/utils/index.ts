import { Category } from '../../category/types';
import { Genre } from '../types';

export const mapGenreToForm = (genre: Genre) => {
  return {
    id: genre.id,
    name: genre.name,
    is_active: genre?.is_active,
    categories_id: genre.categories?.map((category: Category) => category.id),
  };
};
