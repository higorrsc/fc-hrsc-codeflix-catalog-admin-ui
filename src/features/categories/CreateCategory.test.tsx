import { renderWithProviders } from '../../utils/test-utils';
import { CreateCategory } from './CreateCategory';

describe('Test CreateCategory page', () => {
  it('should render correctly', () => {
    const { asFragment } = renderWithProviders(<CreateCategory />);
    expect(asFragment()).toMatchSnapshot();
  });
});
