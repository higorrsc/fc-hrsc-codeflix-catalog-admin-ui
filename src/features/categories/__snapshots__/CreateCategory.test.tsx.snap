// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test CreateCategory page should render correctly 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-0"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-mkslur-MuiPaper-root"
      style="--Paper-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12); --Paper-overlay: linear-gradient(rgba(255, 255, 255, 0.051), rgba(255, 255, 255, 0.051));"
    >
      <div
        class="MuiBox-root css-19midj6"
      >
        <div
          class="MuiBox-root css-1qm1lh"
        >
          <h4
            class="MuiTypography-root MuiTypography-h4 css-f84ov1-MuiTypography-root"
          >
            Create Category
          </h4>
        </div>
      </div>
      <div
        class="MuiBox-root css-0"
      >
        <div
          class="MuiBox-root css-19midj6"
        >
          <form>
            <div
              class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-row MuiGrid-spacing-xs-3 css-1u8xl8z-MuiGrid-root"
            >
              <div
                class="MuiGrid-root MuiGrid-direction-xs-row MuiGrid-grid-xs-12 css-yqiqnf-MuiGrid-root"
              >
                <div
                  class="MuiFormControl-root MuiFormControl-fullWidth css-ytlejw-MuiFormControl-root"
                >
                  <div
                    class="MuiFormControl-root MuiTextField-root css-1xp5r68-MuiFormControl-root-MuiTextField-root"
                  >
                    <label
                      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-1jezcgs-MuiFormLabel-root-MuiInputLabel-root"
                      data-shrink="false"
                      for="«r0»"
                      id="«r0»-label"
                    >
                      Name
                      <span
                        aria-hidden="true"
                        class="MuiFormLabel-asterisk MuiInputLabel-asterisk css-mzcsug-MuiFormLabel-asterisk"
                      >
                         *
                      </span>
                    </label>
                    <div
                      class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-11sqt1k-MuiInputBase-root-MuiOutlinedInput-root"
                    >
                      <input
                        aria-invalid="false"
                        class="MuiInputBase-input MuiOutlinedInput-input css-j3jqhj-MuiInputBase-input-MuiOutlinedInput-input"
                        data-testid="name-input"
                        id="«r0»"
                        name="name"
                        required=""
                        type="text"
                        value=""
                      />
                      <fieldset
                        aria-hidden="true"
                        class="MuiOutlinedInput-notchedOutline css-12rqw02-MuiOutlinedInput-notchedOutline"
                      >
                        <legend
                          class="css-81qg8w"
                        >
                          <span>
                            Name *
                          </span>
                        </legend>
                      </fieldset>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="MuiGrid-root MuiGrid-direction-xs-row MuiGrid-grid-xs-12 css-yqiqnf-MuiGrid-root"
              >
                <div
                  class="MuiFormControl-root MuiFormControl-fullWidth css-ytlejw-MuiFormControl-root"
                >
                  <div
                    class="MuiFormControl-root MuiTextField-root css-1xp5r68-MuiFormControl-root-MuiTextField-root"
                  >
                    <label
                      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-1jezcgs-MuiFormLabel-root-MuiInputLabel-root"
                      data-shrink="false"
                      for="«r1»"
                      id="«r1»-label"
                    >
                      Description
                      <span
                        aria-hidden="true"
                        class="MuiFormLabel-asterisk MuiInputLabel-asterisk css-mzcsug-MuiFormLabel-asterisk"
                      >
                         *
                      </span>
                    </label>
                    <div
                      class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-11sqt1k-MuiInputBase-root-MuiOutlinedInput-root"
                    >
                      <input
                        aria-invalid="false"
                        class="MuiInputBase-input MuiOutlinedInput-input css-j3jqhj-MuiInputBase-input-MuiOutlinedInput-input"
                        data-testid="description-input"
                        id="«r1»"
                        name="description"
                        required=""
                        type="text"
                        value=""
                      />
                      <fieldset
                        aria-hidden="true"
                        class="MuiOutlinedInput-notchedOutline css-12rqw02-MuiOutlinedInput-notchedOutline"
                      >
                        <legend
                          class="css-81qg8w"
                        >
                          <span>
                            Description *
                          </span>
                        </legend>
                      </fieldset>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="MuiGrid-root MuiGrid-direction-xs-row MuiGrid-grid-xs-12 css-yqiqnf-MuiGrid-root"
              >
                <div
                  class="MuiFormGroup-root css-1mac7pt-MuiFormGroup-root"
                >
                  <label
                    class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-195ii7s-MuiFormControlLabel-root"
                  >
                    <span
                      class="MuiSwitch-root MuiSwitch-sizeMedium css-rehya9-MuiSwitch-root"
                    >
                      <span
                        class="MuiButtonBase-root MuiSwitch-switchBase MuiSwitch-colorSecondary PrivateSwitchBase-root MuiSwitch-switchBase MuiSwitch-colorSecondary css-w758z9-MuiButtonBase-root-MuiSwitch-switchBase"
                        data-testid="is-active-input"
                      >
                        <input
                          aria-label="controlled"
                          class="PrivateSwitchBase-input MuiSwitch-input css-j8yymo"
                          name="is_active"
                          type="checkbox"
                        />
                        <span
                          class="MuiSwitch-thumb css-17jyosd-MuiSwitch-thumb"
                        />
                      </span>
                      <span
                        class="MuiSwitch-track css-xz0t6b-MuiSwitch-track"
                      />
                    </span>
                    <span
                      class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-rizt0-MuiTypography-root"
                    >
                      Active
                    </span>
                  </label>
                </div>
              </div>
              <div
                class="MuiGrid-root MuiGrid-direction-xs-row MuiGrid-grid-xs-12 css-yqiqnf-MuiGrid-root"
              >
                <div
                  class="MuiBox-root css-10egq61"
                >
                  <a
                    class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary css-bpairl-MuiButtonBase-root-MuiButton-root"
                    href="/categories"
                    tabindex="0"
                  >
                    Back
                  </a>
                  <button
                    class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorSecondary MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorSecondary css-1mffwr6-MuiButtonBase-root-MuiButton-root"
                    tabindex="0"
                    type="submit"
                  >
                    Save
                  </button>
                </div>
              </div>
            </div>
          </form>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
